src/common-ssh/ssh.c

```diff
@@ -412,6 +428,84 @@ static int guac_common_ssh_authenticate(guac_common_ssh_session* common_session)
 
 }
 
+/**
+ * Verifies if given algorithms are supported by libssh2 and
+ * makes the |prefs| string to use with libssh2_session_method_pref().
+ * 
+ * If the verification fails, all preferred algorithms |algs| are
+ * added to the |prefs| string.
+ *
+ * @param client
+ *     The Guacamole client that is using SSH.
+ * 
+ * @param session
+ *     The session associated with the user to be authenticated.
+ *
+ * @param method_type
+ *      One of the libssh2 Method Type constants for libssh2_session_method_pref().
+ * 
+ * @param algs
+ *      A preferred list of algorithms, for example FIPS_COMPLIANT_CIPHERS.
+ * 
+ * @return
+ *     The |prefs| string to use with libssh2_session_method_pref().
```

Comment:
The fact that this pointer must eventually be freed with a call to `guac_mem_free()` should be documented.